### PR TITLE
AI Analytics Functionality Refactor

### PR DESCRIPTION
**Pull request by tarecord**
This improves the locations of Analytics specific functionality so that it's all encapsulated within a single class. It also simplifies calling the `sendEvent()` method in JS.

### CODE REVIEW

File: src/extension/analytics/send-event.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ *
+ * @param {string} event_label
+ * @param {object} event_data
+ *
+ * @return {Promise<object>} Promise returns object
+ */
+export async function sendEvent(event_label, event_data) {
+	try {
+		return await apiFetch({
+			path: '/kb-design-library/v1/handle_event?XDEBUG_SESSION_START',
```
Comment (defunctl) [inline src/extension/analytics/send-event.js:14]: Did you mean to leave this in `XDEBUG_SESSION_START`?

File: includes/class-kadence-blocks-ai-events.php [hunk 2]
```diff
@@ -16,6 +16,16 @@
  */
 class Kadence_Blocks_AI_Events {
 
+	/**
+	 * Event Label.
+	 */
+	const PROP_EVENT_LABEL = 'event_label';
```
Comment (defunctl) [inline includes/class-kadence-blocks-ai-events.php:22]:
```suggestion
	public const PROP_EVENT_LABEL = 'event_label';
```

File: includes/class-kadence-blocks-ai-events.php [hunk 3]
```diff
@@ -16,6 +16,16 @@
...
+
+	/**
+	 * Event Value.
+	 */
+	const PROP_EVENT_DATA = 'event_data';
```
Comment (defunctl) [inline includes/class-kadence-blocks-ai-events.php:27]:
```suggestion
	public const PROP_EVENT_DATA = 'event_data';
```

File: includes/class-kadence-blocks-ai-events.php [hunk 4]
```diff
@@ -35,11 +45,57 @@ class Kadence_Blocks_AI_Events {
...
+			)
+		);
+	}
+
+	/**
+	 * Checks if a given request has access to search content.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 *
+	 * @return bool|WP_Error True if the request has search access, WP_Error object otherwise.
```
Comment (defunctl) [inline includes/class-kadence-blocks-ai-events.php:88]: I don't think this ever returns a WP_Error

File: src/extension/analytics/send-event.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import apiFetch from '@wordpress/api-fetch';
+
+/**
+ * Send Event to Backend.
+ *
+ * @param {string} event_label
+ * @param {object} event_data
+ *
+ * @return {Promise<object>} Promise returns object
```
Comment (defunctl) [inline src/extension/analytics/send-event.js:9]:
```suggestion
 * @return {string | Promise<object>} Promise returns object
```

**Review by defunctl — CHANGES_REQUESTED** [review]

File: src/extension/analytics/send-event.js (hunk 1 above)
Comment (tarecord) [reply src/extension/analytics/send-event.js:14]: Ugh, no 🤦🏼‍♂️ thanks!

File: includes/class-kadence-blocks-ai-events.php [hunk 6]
```diff
@@ -35,11 +45,55 @@ class Kadence_Blocks_AI_Events {
...
+			)
+		);
+	}
+
+	/**
+	 * Checks if the current user has access to edit posts.
+	 *
+	 * @return bool
+	 */
+	public function verify_user_can_edit() {
```
Comment (defunctl) [inline includes/class-kadence-blocks-ai-events.php:88]:
```suggestion
	public function verify_user_can_edit(): bool {
```

File: includes/class-kadence-blocks-ai-events.php [hunk 7]
```diff
@@ -110,4 +164,71 @@ public static function get_prophecy_token_header( $args = [] ) {
...
 		return base64_encode( json_encode( $parsed_args ) );
 	}
+
+	/**
+	 * Configures various event requests to the /analytics/event endpoint
+	 * and sends them to ProphecyWP.
+	 *
+	 * @param WP_REST_Request $request The request to the endpoint.
+	 */
+	public function handle_event_endpoint( $request ) {
```
Comment (defunctl) [inline includes/class-kadence-blocks-ai-events.php:174]:
```suggestion
	public function handle_event_endpoint( WP_REST_Request $request ): WP_REST_Response  {
```

**Review by defunctl — CHANGES_REQUESTED** [review]
Just some type hinting and this is good to go on my end.

**Review by defunctl — APPROVED** [review]

**Pull request merged**